tekton-catalog/any-sequencer/cmd/root.go

```diff
@@ -0,0 +1,246 @@
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func checkConditions(crs []conditionResult, tr *v1beta1.TaskRun) (string, bool) {
+	for _, cr := range crs {
+		parameters := make(map[string]interface{})
+		for _, result := range cr.results {
+			trLabel := tr.Labels["tekton.dev/pipelineTask"]
+			var found bool
+			for _, taskRunResults := range tr.Status.TaskRunResults {
+				if result == taskRunResults.Name {
+					parameters[`results_`+trLabel+`_`+result] = taskRunResults.Value
+					found = true
+					break
+				}
+			}
+			if !found {
+				log.Printf("The result %s does not exist in taskrun %s.", result, trLabel)
+				return cr.condition, false
+			}
+		}
+		expr, err := govaluate.NewEvaluableExpression(cr.condition)
+		if err != nil {
+			log.Fatal("syntax error:", err)
+		}
+
+		evaluateresult, err := expr.Evaluate(parameters)
+		if err != nil {
+			log.Fatal("evaluate error:", err)
+		}
+
+		if result, ok := evaluateresult.(bool); ok {
+			if result {
+				continue
+			}
+		}
+		return cr.condition, false
+	}
+	return "", true
+}
+
+func watch(cmd *cobra.Command, args []string) {
+	log.Printf("Starting to watch taskrun for '%s' and condition in %s/%s.", taskList, namespace, prName)
+
+	tasks := strings.Split(taskList, ",")
+
+	parse_conditions(conditions, &tasks)
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Errorf("Get config of the cluster failed: %+v", err)
+		os.Exit(1)
+	}
+
+	tektonClient, err := tektoncdclientset.NewForConfig(config)
+	if err != nil {
+		log.Errorf("Get client of tekton failed: %+v", err)
+		os.Exit(1)
+	}
+
+	labelSelector := "tekton.dev/pipelineRun=" + strings.TrimSpace(prName)
+
+	var failedTasks []string
+	for {
+		watcher, err := tektonClient.TektonV1beta1().TaskRuns(namespace).Watch(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+
+		if err != nil {
+			log.Printf("TaskRun Watcher error:" + err.Error())
+			log.Printf("Please ensure the service account has permission to get taskRun.")
+			os.Exit(1)
+		}
+
+		for event := range watcher.ResultChan() {
+			taskrun := event.Object.(*v1beta1.TaskRun)
+			taskLabel := taskrun.Labels["tekton.dev/pipelineTask"]
+			if contains(tasks, taskLabel) {
+				if taskrun.Status.GetCondition(apis.ConditionSucceeded).IsTrue() {
+					log.Printf("The TaskRun of %s succeeded.", taskLabel)
+					// check the conditions for the task
+					if conditions, ok := conditionMap[taskLabel]; ok {
+						if condition, ok := checkConditions(conditions, taskrun); !ok {
+							log.Printf("The condition %s for the task %s does not meet, still waiting", condition, taskLabel)
+							failedTasks = append(failedTasks, taskLabel)
+						} else {
+							watcher.Stop()
+							os.Exit(0)
+						}
+					} else {
+						watcher.Stop()
+						os.Exit(0)
+					}
+				} else if taskrun.Status.GetCondition(apis.ConditionSucceeded).IsFalse() {
+					if !contains(failedTasks, taskLabel) {
+						failedTasks = append(failedTasks, taskLabel)
```

Comment:
if so, the `taskrun` and `taskrun.condition` will be `all` relationship, but not any. I'm OK if this is design.

Reply:
In fact the taskrun and the condition on the same taskrun is of ```all``` relationship. That's by design.